specification/dartLangSpec.tex

```diff
@@ -14081,6 +14084,23 @@ \subsection{Property Extraction}
 (\ref{superGetterAccessAndMethodClosurization}).
 \EndCase
 
+\LMHash{}%
+\Case{Implicit}
+Let $e$ be an expression whose static type is
+an interface type that has a method named \CALL.
+In the case where the context type for $e$ is a function type,
+$e$ is treated as \code{$e$.\CALL}.
```

Comment:
I want to have a consistent terminology for desugaring (and 'desugar' seems perhaps a bit too informal, even though it would probably offer very good readability).

However, we're not there yet: Old parts of the specification use 'is equivalent to', but it is a considerable unnecessary complication if we start implying that desugaring is a relation (so we may need to go back and forth, searching for a "solution" of sorts in a large space of possible rewritings), I want this to be a simple function: Take the given source code, desugar it, and that's the foundation for further work.

The phrase I've used (in many places, at least 63 places right now)  is 'A is treated as B'. At least, this is directional.

(Vague definitions mean exactly what's the most convenient thing right now ;-)